solr/solrj/src/java/org/apache/solr/common/util/CommonTestInjection.java

```diff
@@ -129,15 +111,61 @@ public static boolean injectBreakpoint(String key, Object... args) {
       log.info("Breakpoint with key {} is triggered", key);
       breakpoint.executeAndResume(args);
       log.info("Breakpoint with key {} was executed and normal code execution resumes", key);
+    } else {
+      log.info(
+          "Breakpoint with key {} is triggered but there's no implementation set. Skipping...",
+          key);
```

Comment:
I don't think we should log here (and fwiw there is no corresponding logging for absent `injectDelay()`. The issue is that if you were to enable assertions on a running system (not in a test context), you might get a ton of spam from this. If you want to leave it, maybe set log level from `log.info()` to `log.debug()`?

Reply:
yes let's change that to debug, i was a bit concerned if someone accidentally used different key names meant for the same Breakpoint for `injectBreakpoint` and `setImplementation`, so with this log message they can see whether the breakpoint is triggered.

Though I guess this could be really noisy as you pointed out, and I assume the dev can search whether there's "Breakpoint with key ... is triggered" message instead and absence of that means either the execution path does not reach the breakpoint or the breakpoint has no implementation set